main.go

```diff
@@ -2,12 +2,32 @@ package main
 
 import (
 	"context"
+	"fmt"
+	"os"
 
-	"github.com/spf13/cobra"
-
+	"github.com/ollama/ollama/api"
 	"github.com/ollama/ollama/cmd"
 )
 
 func main() {
-	cobra.CheckErr(cmd.NewCLI().ExecuteContext(context.Background()))
+	checkErr(cmd.NewCLI().ExecuteContext(context.Background()))
+}
+
+// checkErr prints the error message and exits the program if the message is not nil.
+func checkErr(msg any) {
+	if msg == nil {
```

Comment:
Might just be me but I'd invert this equality

```go
if err != nil {
  ...

  os.Exit(1)
}
```

Reply:
I ended up keeping this part, just because inverting the equality indents all the logic in the function, making it kinda weird looking